Fix java Xlint warnings

These warnings print every time gradle builds, so it would be nice to fix them to avoid noisy builds (and perhaps to ensure code correctness?):

```
Users/dane/projects/mlt2/java/src/main/java/com/mlt/converter/mvt/MvtUtils.java:105: warning: [unchecked] unchecked cast
        var properties = ((LinkedHashMap<String, Object>) mvtFeature.getUserData());
                                                                                ^
  required: LinkedHashMap<String,Object>
  found:    Object
/Users/dane/projects/mlt2/java/src/main/java/com/mlt/decoder/MltDecoder.java:83: warning: [unchecked] unchecked cast
            var p = ((Map<String, Object>) propertyColumn);
                                           ^
  required: Map<String,Object>
  found:    Object
/Users/dane/projects/mlt2/java/src/main/java/com/mlt/decoder/MltDecoder.java:85: warning: [unchecked] unchecked cast
              properties.put(a.getKey(), (List<Object>) a.getValue());
                                                                  ^
  required: List<Object>
  found:    Object
/Users/dane/projects/mlt2/java/src/main/java/com/mlt/decoder/MltDecoder.java:88: warning: [unchecked] unchecked method invocation: method put in class HashMap is applied to given types
            properties.put(columnName, (ArrayList) propertyColumn);
                          ^
  required: K,V
  found:    String,ArrayList
  where K,V are type-variables:
    K extends Object declared in class HashMap
    V extends Object declared in class HashMap
/Users/dane/projects/mlt2/java/src/main/java/com/mlt/decoder/MltDecoder.java:88: warning: [unchecked] unchecked conversion
            properties.put(columnName, (ArrayList) propertyColumn);
                                       ^
  required: V
  found:    ArrayList
  where V is a type-variable:
    V extends Object declared in class HashMap
15 warnings

```

--- a/java/src/main/java/com/mlt/converter/mvt/MvtUtils.java
+++ b/java/src/main/java/com/mlt/converter/mvt/MvtUtils.java
@@ -102,7 +102,8 @@ public class MvtUtils {
       var mvtFeatures = layer.getGeometries();
       var features = new ArrayList<Feature>();
       for (var mvtFeature : mvtFeatures) {
-        var properties = ((LinkedHashMap<String, Object>) mvtFeature.getUserData());
+        Map<?, ?> properties =
+            mvtFeature.getUserData() instanceof Map<?, ?> map ? map : new LinkedHashMap<>();
         var id = (long) properties.get(ID_KEY);
         properties.remove(ID_KEY);
         // TODO: quick and dirty -> implement generic
@@ -117,10 +118,9 @@ public class MvtUtils {
     return new MapboxVectorTile(layers);
   }
 
-  private static LinkedHashMap<String, Object> transformNestedPropertyNames(
-      Map<String, Object> properties) {
+  private static Map<String, Object> transformNestedPropertyNames(Map<?, ?> properties) {
     var transformedProperties = new LinkedHashMap<String, Object>();
-    properties.forEach((k, v) -> transformedProperties.put(k.replace("_", ":"), v));
+    properties.forEach((k, v) -> transformedProperties.put(k.toString().replace("_", ":"), v));
     return transformedProperties;
   }
 }

--- a/java/src/main/java/com/mlt/decoder/MltDecoder.java
+++ b/java/src/main/java/com/mlt/decoder/MltDecoder.java
@@ -17,7 +17,6 @@ import com.mlt.vector.flat.LongFlatVector;
 import com.mlt.vector.geometry.GeometryVector;
 import java.io.IOException;
 import java.util.*;
-import java.util.stream.Collectors;
 import me.lemire.integercompression.IntWrapper;
 import org.locationtech.jts.geom.Geometry;
 
@@ -35,7 +34,7 @@ public class MltDecoder {
     while (offset.get() < tile.length) {
       List<Long> ids = null;
       Geometry[] geometries = null;
-      var properties = new HashMap<String, List<Object>>();
+      var properties = new HashMap<String, List<?>>();
 
       var version = tile[offset.get()];
       offset.increment();
@@ -69,7 +68,7 @@ public class MltDecoder {
                 IntegerDecoder.decodeIntStream(tile, offset, idDataStreamMetadata, false).stream()
                     .mapToLong(i -> i)
                     .boxed()
-                    .collect(Collectors.toList());
+                    .toList();
           } else {
             ids = IntegerDecoder.decodeLongStream(tile, offset, idDataStreamMetadata, false);
           }
@@ -79,13 +78,14 @@ public class MltDecoder {
         } else {
           var propertyColumn =
               PropertyDecoder.decodePropertyColumn(tile, offset, columnMetadata, numStreams);
-          if (propertyColumn instanceof HashMap<?, ?>) {
-            var p = ((Map<String, Object>) propertyColumn);
-            for (var a : p.entrySet()) {
-              properties.put(a.getKey(), (List<Object>) a.getValue());
+          if (propertyColumn instanceof Map<?, ?> map) {
+            for (var a : map.entrySet()) {
+              properties.put(
+                  a.getKey().toString(),
+                  a.getValue() instanceof List<?> list ? list : List.of(a.getValue()));
             }
-          } else {
-            properties.put(columnName, (ArrayList) propertyColumn);
+          } else if (propertyColumn instanceof List<?> list) {
+            properties.put(columnName, list);
           }
         }
       }
@@ -179,7 +179,7 @@ public class MltDecoder {
   private static Layer convertToLayer(
       List<Long> ids,
       Geometry[] geometries,
-      Map<String, List<Object>> properties,
+      Map<String, List<?>> properties,
       MltTilesetMetadata.FeatureTableSchema metadata,
       int numFeatures) {
     if (numFeatures != geometries.length || numFeatures != ids.size()) {

